src/util/manifest.js

```diff
@@ -33,17 +33,20 @@ export default function getValidatedManifest(sourceDir: string): Promise {
         errors.push('missing "version" property');
       }
 
-      // Make sure the manifest defines a gecko id.
-      let idProps = ['applications', 'gecko', 'id'];
-      var propPath = '';
-      var objectToCheck = manifestData;
-
-      for (let nextProp of idProps) {
-        propPath = propPath ? `${propPath}.${nextProp}` : nextProp;
-        objectToCheck = objectToCheck[nextProp];
-        if (!objectToCheck) {
-          errors.push(`missing "${propPath}" property`);
-          break;
+      if (manifestData.applications) {
```

Comment:
@kumar303 I'm assuming that it is possible that we want to omit the `id` and, at the same time, we want the `applications` section to enforce the `strict_min_version`, `strict_max_version` (or probably, even if it is rarely used, the `update_url`).

If the above assumption is true, we should change this validation accordingly.